datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1222,119 +1222,144 @@ array_removement_function!(
     "Array_remove_all SQL function"
 );
 
-fn general_replace(args: &[ArrayRef], arr_n: Vec<i64>) -> Result<ArrayRef> {
-    let list_array = as_list_array(&args[0])?;
-    let from_array = &args[1];
-    let to_array = &args[2];
-
+/// For each element of `list_array[i]`, replaces up to `arr_n[i]`  occurences
```

Comment:
In general, having a clear description of what the function does, especially when it is mind bending like`general_replace` helps a lot. I also find that writing such a description often helps improve the code as well as results in better naming

For example, I struggled to explain how this function worked when it took `&[ArrayRef]` because then the explanations were in terms fof `arg[0]`, `arg[1]`, and `arg[2]`. Giving the parameters names made things much clearer